cli/testflinger_cli/__init__.py

```diff
@@ -317,8 +373,16 @@ def submit(self):
                 raise SystemExit(
                     "File not found: {}".format(self.args.filename)
                 ) from exc
-        job_id = self.submit_job_data(data)
-        queue = yaml.safe_load(data).get("job_queue")
+        job_dict = yaml.safe_load(data)
+
+        archive_path = self.pack_attachments(job_dict)
+        job_id = self.submit_job_data(job_dict)
+        if archive_path is not None:
+            # follow job submission with attachment submission, if required
+            self.submit_job_attachments(job_id, archive_path)
```

Comment:
I'm thinking that we may want to handle failures with submit_job_attachments a bit differently. If we have it detect http errors and exit, then we've already transmitted the job_data and it's just stuck in a waiting state for the attachments. Eventually, I know the db ttl should clean it up, but in the meantime, users would just see that their job is out there waiting to run but never does.
Perhaps we should retry here, and if we still don't succeed, then attempt to cancel the job and warn the user that it was unsuccessful before exiting. What do you think?